api/src/main/java/org/apache/iceberg/FileScanTask.java

```diff
@@ -26,15 +26,21 @@
  * A scan task over a range of a single file.
  */
 public interface FileScanTask extends ScanTask {
+
+  /**
+   * @return the file to scan
+   */
+  ContentFile<?> contentFile();
```

Comment:
This PR raises a somehow similar question about the future of `FileScanTask` as in #4870. Maybe, we can discuss there a bit.

Instead of adding a new method like `contentFile`, what about wrapping a delete file as `DataFile` and using the task without any changes?

Reply:
Yea I think that works although it felt not so clean,  DeleteDataFile?   It will definitely be less changes in code, I'm ok though if everyone agrees.  Will try to follow there as well

Reply:
@aokolnychyi reverted the refactor to move this forward.  As over there it seems there is no need to represent EqualityDelete as DataFile, it seems this becomes the only weird case (outside of course the metadata tables).  If we want to do it later, its still open.